app/Service/User/UserRegisterService.php

```diff
@@ -7,43 +7,71 @@
 use App\Models\Resume;
 use App\Models\Student;
 use App\Models\User;
-use Illuminate\Support\Facades\DB;
 use Exception;
 use PDOException;
+use Error;
 
 class UserRegisterService
 {
-    public function createUser(array $input): array | bool
+    public function createUser(array $input): array
     {
-        if (empty($input['email']) || empty($input['password'])) {
-            return false;
-        }
-
         $input['password'] = bcrypt($input['password']);
 
-        DB::beginTransaction();
+        $user = $this->createUserRecord($input);
+        // var_dump($user->id);exit;
+        $student = $this->createStudentRecord($user->id);
+        $this->createResumeRecord($student->id, $input['specialization'] ?? null);
+        $success['token'] = $this->generateAccessToken($user);
+        $success['email'] = $user->email;
 
+        return $success;
+    }
+
+    private function createUserRecord(array $input): User
+    {
         try {
-            $user = User::create($input);
+            return User::create($input);
+        } catch (PDOException $e) {
+            throw new Exception('Error al crear el usuario en la base de datos: ' . $e->getMessage());
+        } catch (Error $e) {
+            throw new Exception('Error inesperado al crear el usuario: ' . $e->getMessage());
```

Comment:
You don't need two catches for this, just take the error and return UserRegisterException for both of them.

Reply:
done